chia/data_layer/data_store.py

```diff
@@ -843,6 +865,98 @@ async def delete(
 
         return
 
+    async def insert_batch(
+        self,
+        tree_id: bytes32,
+        changelist: List[Dict[str, Any]],
+        status: Status = Status.PENDING,
+        *,
+        lock: bool = True,
+    ) -> None:
+        async with self.db_wrapper.locked_transaction(lock=lock):
+            hint_keys_values = await self.get_keys_values_dict(tree_id, lock=False)
```

Comment:
yes, all terminal nodes will be in memory. this is still old code - I've just moved it from data_layer to here, since it looked more appropriate, so it worked like this before. the point is to optimise looking up a value by key - needed by `delete`. without this we had very slow performance.